mock_drivers/cpp_mock_drivers/src/MockCANDriver.cpp

```diff
@@ -0,0 +1,225 @@
+
+    }
+
+    int MockCANDriver::run(){
+
+        mock_driver_node_.init();
+
+        // bar parser subscriber
+        mock_driver_node_.addSub<boost::shared_ptr<ROSComms<const carma_simulation_msgs::BagData::ConstPtr&>>>(bag_parser_sub_ptr_);
+
+        // data topic publishers
+        mock_driver_node_.addPub<boost::shared_ptr<ROSComms<std_msgs::Bool>>>(acc_engaged_ptr_);
+        mock_driver_node_.addPub<boost::shared_ptr<ROSComms<std_msgs::Float64>>>(acceleration_ptr_);
+        mock_driver_node_.addPub<boost::shared_ptr<ROSComms<std_msgs::Bool>>>(antilock_brakes_active_ptr_);
+        mock_driver_node_.addPub<boost::shared_ptr<ROSComms<std_msgs::Bool>>>(brake_lights_ptr_);
+        mock_driver_node_.addPub<boost::shared_ptr<ROSComms<std_msgs::Float64>>>(brake_position_ptr_);
+        mock_driver_node_.addPub<boost::shared_ptr<ROSComms<std_msgs::Float64>>>(engine_speed_ptr_);
+        mock_driver_node_.addPub<boost::shared_ptr<ROSComms<std_msgs::Float64>>>(fuel_flow_ptr_);
+        mock_driver_node_.addPub<boost::shared_ptr<ROSComms<std_msgs::Float64>>>(odometer_ptr_);
+        mock_driver_node_.addPub<boost::shared_ptr<ROSComms<std_msgs::Bool>>>(parking_brake_ptr_);
+        mock_driver_node_.addPub<boost::shared_ptr<ROSComms<std_msgs::Float64>>>(speed_ptr_);
+        mock_driver_node_.addPub<boost::shared_ptr<ROSComms<std_msgs::Bool>>>(stability_ctrl_active_ptr_);
+        mock_driver_node_.addPub<boost::shared_ptr<ROSComms<std_msgs::Bool>>>(stability_ctrl_enabled_ptr_);
+        mock_driver_node_.addPub<boost::shared_ptr<ROSComms<std_msgs::Float64>>>(steering_wheel_angle_ptr_);
+        mock_driver_node_.addPub<boost::shared_ptr<ROSComms<std_msgs::Float64>>>(throttle_position_ptr_);
+        mock_driver_node_.addPub<boost::shared_ptr<ROSComms<std_msgs::Bool>>>(traction_ctrl_active_ptr_);
+        mock_driver_node_.addPub<boost::shared_ptr<ROSComms<std_msgs::Bool>>>(traction_ctrl_enabled_ptr_);
+        mock_driver_node_.addPub<boost::shared_ptr<ROSComms<j2735_msgs::TransmissionState>>>(transmission_state_ptr_);
+        mock_driver_node_.addPub<boost::shared_ptr<ROSComms<cav_msgs::TurnSignal>>>(turn_signal_state_ptr_);
+        mock_driver_node_.addPub<boost::shared_ptr<ROSComms<geometry_msgs::TwistStamped>>>(vehicle_twist_ptr_);
+        mock_driver_node_.addPub<boost::shared_ptr<ROSComms<autoware_msgs::VehicleStatus>>>(vehicle_status_ptr_);
+        mock_driver_node_.addPub<boost::shared_ptr<ROSComms<automotive_platform_msgs::VelocityAccel>>>(velocity_accel_ptr_);
+
+        // driver discovery publisher
+        mock_driver_node_.addPub<boost::shared_ptr<ROSComms<cav_msgs::DriverStatus>>>(driver_discovery_pub_ptr_);
+        mock_driver_node_.setSpinCallback(std::bind(&MockCANDriver::driverDiscovery, this));
+
+        mock_driver_node_.spin(1);
```

Comment:
This data rate is far to slow. The callback for the bag data won't be triggered often enough.